Sources/NIOPosix/NonBlockingFileIO.swift

```diff
@@ -627,4 +627,150 @@ public struct NonBlockingFileIO: NIOSendable {
         }
     }
 
+#if !os(Windows)
+    /// Returns information about a file at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// - note: If `path` is a symlink, information about the link, not the file it points to.
+    ///
+    /// - parameters:
+    ///     - path: The path of the file to get information about.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` containing file information.
+    public func lstat(path: String, eventLoop: EventLoop) -> EventLoopFuture<stat> {
+        return self.threadPool.runIfActive(eventLoop: eventLoop) {
+            var s = stat()
+            try Posix.lstat(pathname: path, outStat: &s)
+            return s
+        }
+    }
+
+    /// Creates a symbolic link to a  `destination` file  at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    ///
+    /// - parameters:
+    ///     - path: The path of the link.
+    ///     - destination: Target path where this link will point to.
+    ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
+    /// - returns: An `EventLoopFuture` which is fulfilled if the rename was successful or fails on error.
+    public func symlink(path: String, to destination: String, eventLoop: EventLoop) -> EventLoopFuture<Void> {
```

Comment:
I wonder if we should stick with close-to-syscall naming or go with foundation-esque naming? `createSymbolicLink`, `attributesOfAnItem` or similar...

Reply:
I'm inclined for us to stick to close-to-syscall naming: this is NIOPosix, after all, we aren't pretending to cross-platform compatibility here.

Reply:
should `listDirectory` be `readdir` then? Although we do have `openFile` and not `open`...

Reply:
I'm open to leaving this as-is.